### DISCUSSION TITLE
Advice on the cleanest/best way to handle converting Decimals to Int during serialization

### DISCUSSION THREAD

**Discussion by bryanprange** [pydantic/pydantic#10295] [Question]
Hi I'm using Pydantic with DynamoDB, and am trying to figure out the best way to serialize Decimals to Integers without throwing: "Expected `int` but got `Decimal` - serialized value may not be as expected"

DynamoDB stores numeric data as type `Number`, which is floating point. The `boto3` DynamoDB deserializer always converts `Number` to the Python `Decimal` type.

In my existing model code, I declare the field as an `int`, which works great except for the above warning log.
```
class My(BaseModel):
    id: int
```

I've found that I can use a `PlainSerializer` to work around the warning, but it seems redundant. 
```
class My(BaseModel):
    id: Annotated[int, PlainSerializer(lambda v: int(v))]
```

I also tried declaring the field as a Decimal with `decimal_places=0`, but I've found that (probably due to the fact that Python treats `5` and `5.0` the same) zero value decimal points will not be removed when constructing the model (`5.0` in, `5.0` out)
```
class My(BaseModel):
    id: Decimal = Field(decimal_places=0)
```

Sorry for the long-winded question, and thanks for any advice.

**Comment by neanias** [comment]
I found an answer to this question here: https://github.com/pydantic/pydantic/issues/360#issuecomment-2621736240

> I am using this approach to allow floats to be converted to integers. `FlexibleInt = Annotated[int, BeforeValidator(lambda x: int(x))]`

I used it like so:

```python
created_at: Annotated[int, BeforeValidator(lambda num: int(num))]
```
